vendor/k8s.io/kube-openapi/pkg/handler/handler.go

```diff
@@ -55,13 +55,38 @@ type OpenAPIService struct {
 
 	lastModified time.Time
 
-	specBytes []byte
-	specPb    []byte
-	specPbGz  []byte
+	openapiSpec *spec.Swagger
+	specBytes   []byte
+	specPb      []byte
+	specPbGz    []byte
 
 	specBytesETag string
 	specPbETag    string
 	specPbGzETag  string
+
+	jcache    jsonCache
+	pbCache   protobufCache
+	pbgzCache protobufGzipCache
+}
+
+type baseCache struct {
+	cacheClean bool
+	cacheBytes []byte
```

Comment:
Maybe you can drop the `cacheClean` param, and set `cacheBytes` to `nil` when the cache is "empty"?